src/main/java/workflow/WorkflowManager.java

```diff
@@ -0,0 +1,20 @@
+package workflow;
+
+import workflow.context.AbstractCompositeContext;
+
+import java.util.HashMap;
+
+public enum WorkflowManager {
+    INSTANCE;
+
+    HashMap<String, AbstractCompositeContext> contextMap = new HashMap<String, AbstractCompositeContext>();
+
+    public AbstractCompositeContext getContextByKey(String key) {
```

Comment:
So we really are making a hash table of our contexts?

Some comments about how this is intended to work are needed.